src/kwio.sh

```diff
@@ -119,3 +119,36 @@ function ask_yN()
     printf '%s\n' '0'
   fi
 }
+
+# Asks for user input
+#
+# @message A string with the message to be displayed to the user.
+# @default_option A string with the default answer of the user (if the user simply
+#                 inputs nothing, this will be considered the answer)
+# @show_default A boolean value that defines if the default value will be printed
+#               to the user. If this value is empty, this will be considered true.
+#
+# Returns: The user answer, guaranteed it is not empty.
+#
+# Note: this function does not verify the given answer. You have to handle this
+# somewhere else.
+
+function ask_with_default()
+{
+  local message="$1"
+  local default_option="$2"
+  local show_default="$3"
+
+  if [[ -z "$show_default" ]]; then
+    message="$message ($default_option): "
+  else
+    message="$message: "
+  fi
+
+  read -r -p "$message" response
```

Comment:
Could you try to add a test that captures the default behavior? I mean, you don't have any test that checks the `$message`.

Reply:
Do you mean to test if the received `$message` is empty or something like that?

Reply:
Try to capture the output of the `read` command and check if the message is the default or not.